Do remote build when builder flag is set

ikaneshiro opened the pull request:
**Description of the Pull Request (PR):**

Title

**This fixes or addresses the following GitHub issues:**

- Fixes #2985 


**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)


Attn: @singularity-maintainers

ikaneshiro committed:
perform remote build if --builder flag is specified

ArangoGutierrez reviewed (approved):
Keep it simple, I like it

mem commented on cmd/internal/cli/build_darwin.go:
```diff
@@ -27,7 +27,7 @@ func run(cmd *cobra.Command, args []string) {
 	}
 
-	if !remote {
+	if !remote && !cmd.Flags().Lookup("builder").Changed {
```
I see what you are doing here: either the use set remote explicitly or they specified builder and therefore set remote implicitly.

In order to reduce maintaining costs and avoid having to check the two things continuously, I think it might be easier to check for builder Changed once, and *set* remote if that's the case. Maybe take preRun out of build_linux.go and build_darwin.go (since it's the same in both) and move it to build.go, and perform this check there:

```
func preRun(...) {
    ...
    if cmd.Flags().Lookup("builder").Changed {
        cmd.Flags().Lookup("remote").Value.Set("true")
    }
}
```

in this way, in the rest of the code you can continue checking only remote, as it is now and you don't have to introduce the explicit handling of the dependency everywhere.

It's also a good idea to update the documentation for the builder flag and add something like "(implies --remote)".

mem commented on cmd/internal/cli/build_linux.go:
```diff
@@ -36,7 +36,7 @@ func run(cmd *cobra.Command, args []string) {
 	}
 
-	if remote {
+	if remote || cmd.Flags().Lookup("builder").Changed {
```
I can see that this is the change the issue is asking for: setting builder implies --remote.

ikaneshiro committed:
Set remote to true in preRun() if builder is changed

	This removes preRun() from build_linux.go and build_darwin.go
	and moves it to build.go since those functions are identical.

mem reviewed (approved):
Thanks!
